### PR TITLE
 feat: Add a command to update test source path

### LINKED ISSUE #470: Run/Debug code lenses don't appear
Starting about a week ago, code lenses for each test, as well as the lens for the test class as a whole, have stopped appearing. Also, no tests appear in the test explorer panel, though this may have been a pre-existing issue – I've never used the test explorer before so I can't say for sure.

Restarting or reloading vscode does not fix the problem, nor does restarting the computer, nor disabling and re-enabling the test runner code lens – or the root vscode code lens – setting (also tried disable+restart and re-enable+restart).

Project is using gradle and tests use junit. Gradle version was upgraded from `2.14.1` to `3.5.1` 17 days ago, but the issue has only started occurring in the last week, so I'm not sure that the gradle upgrade is relevant to the problem.

Has anyone else experienced similar issues recently and does anyone have any ideas as to what might be causing the issue?

### PR DESCRIPTION
**Pull request by jdneo**
Resolve #470

demo

### CODE REVIEW

File: java-extension/com.microsoft.java.test.plugin/src/main/java/com/microsoft/java/test/plugin/model/Result.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
+/*******************************************************************************
+* Copyright (c) 2018 Microsoft Corporation and others.
```
Comment (Eskibear) [inline java-extension/com.microsoft.java.test.plugin/src/main/java/com/microsoft/java/test/plugin/model/Result.java]: year 2019

File: package.nls.json [hunk 2]
```diff
@@ -10,7 +10,7 @@
     "contributes.commands.java.test.cancel.title": "Cancel Test Job",
     "contributes.commands.java.test.explorer.refresh.title": "Refresh",
     "contributes.commands.java.test.config.migrate.title": "Migrate Deprecated 'launch.test.json'",
-    "contributes.commands.java.test.listTestSourcePaths.title": "List all Java test source paths",
+    "contributes.commands.java.test.listTestSourcePaths.title": "Update project test source paths",
```
Comment (Eskibear) [inline package.nls.json]: the id `listTestSourcePaths` should be also changed

File: package.nls.zh.json [hunk 3]
```diff
@@ -10,7 +10,7 @@
     "contributes.commands.java.test.cancel.title": "取消测试任务",
     "contributes.commands.java.test.explorer.refresh.title": "刷新",
     "contributes.commands.java.test.config.migrate.title": "迁移已弃用的 'launch.test.json' 文件",
-    "contributes.commands.java.test.listTestSourcePaths.title": "显示测试源文件所在路径",
+    "contributes.commands.java.test.listTestSourcePaths.title": "更新项目测试文件路径",
```
Comment (Eskibear) [inline package.nls.zh.json]: change `listTestSourcePaths`

File: src/commands/testPathCommands.ts [hunk 4]
```diff
@@ -1,8 +1,11 @@
...
 
-import { window, workspace, WorkspaceFolder } from 'vscode';
-import { getTestSourcePaths } from '../utils/commandUtils';
+import { QuickPickItem, Uri, window, workspace, WorkspaceFolder } from 'vscode';
+import { testCodeLensProvider } from '../codeLensProvider';
+import { testExplorer } from '../explorer/testExplorer';
+import { testFileWatcher } from '../testFileWatcher';
+import { getTestSourcePaths, updateTestSourcePaths } from '../utils/commandUtils';
 
 export async function listTestSourcePaths(): Promise<void> {
```
Comment (Eskibear) [inline src/commands/testPathCommands.ts]: function name not match the body

File: src/testFileWatcher.ts [hunk 5]
```diff
@@ -1,46 +1,62 @@
...
 
-    private registerWatcherListeners(watcher: FileSystemWatcher, disposables: Disposable[]): void {
-        watcher.onDidChange((uri: Uri) => {
-            const node: TestTreeNode | undefined = explorerNodeManager.getNode(uri.fsPath);
-            testExplorer.refresh(node);
-        }, null, disposables);
+    public dispose(): void {
+        for (const disposable of this.disposables) {
+            if (disposable) {
+                disposable.dispose();
```
Comment (Eskibear) [inline src/testFileWatcher.ts]: should also delete disposable

**Comment by yaohaizh** [comment]
What is feat?

**Comment by jdneo** [comment]
> What is feat?

Short for `feature`

**Comment by yaohaizh** [comment]
> > What is feat?
> 
> Short for `feature`

I know. The point is that  you should not use abbreviation in the commit message when do squash.

**Comment by jdneo** [comment]
Close for now since it seems to have some other better solutions for this issue.

**Pull request closed without merging**